src/crypto/sodium.cpp

```diff
@@ -22,138 +22,190 @@
 
 #include "mega.h"
 
-#ifdef USE_SODIUM
-namespace mega {
+namespace mega
+{
 
-CryptoPP::AutoSeededRandomPool EdDSA::rng;
+const std::string EdDSA::TLV_KEY = "prEd255";
 
-// Initialise libsodium crypto system.
-void EdDSA::init() {
-    sodium_init();
-}
+EdDSA::EdDSA(unsigned char *keySeed)
+{
+    if (sodium_init() == -1)
+    {
+        LOG_err << "Cannot initialize sodium library.";
+        return;
+    }
 
+    if (keySeed)    // then use the value
+    {
+        memcpy(this->keySeed, keySeed, EdDSA::SEED_KEY_LENGTH);
+    }
+    else    // make the new key seed.
+    {
+        PrnGen::genblock(this->keySeed, EdDSA::SEED_KEY_LENGTH);
+    }
 
-// Sets a private key seed from a buffer.
-void EdDSA::setKeySeed(const char* data) {
-    memcpy(this->keySeed, data, crypto_sign_SEEDBYTES);
+    // derive public and private keys from the seed
+    if (crypto_sign_seed_keypair(this->pubKey, this->privKey, this->keySeed) != 0)
+    {
```

Comment:
Since we are checking if the initialization is correct, we should avoid posterior usages of this object, maybe setting a flag or something like that.